app/utils/YarnLockRepositoryParser.scala

```diff
@@ -35,6 +35,8 @@ object YarnLockRepositoryParser extends AbstractParser {
     extractFromFile(buildFile, yarnVersion, extractProperties)
       .filter(p => npmArtifacts.contains(p._1))
       .map(p => NodeDependency(p._1, p._2, subfolder))
+      .groupBy(_.name)
+      .map(_._2.maxBy(_.version)(VersionComparator))
```

Comment:
as seen together I'm not sure this will work.
A counter-example in an Angular project would be:
```js
// package.json
dependencies: {
   core.js: "^2.5.7"
}
```
producing this yarn.lock:
```
core-js@3.1.4:
  version "3.1.4"
  resolved "https://nexus.wired-aws.com/repository/npm-group/core-js/-/core-js-3.1.4.tgz#3a2837fc48e582e1ae25907afcd6cf03b0cc7a07"
  integrity sha512-YNZN8lt82XIMLnLirj9MhKDFZHalwzzrL9YLt6eb0T5D0EDl4IQ90IGkua8mHbnxNrkj1d8hbdizMc0Qmg1WnQ==

core-js@^2.2.0, core-js@^2.4.0:
  version "2.5.7"
  resolved "https://nexus.wired-aws.com/repository/npm-group/core-js/-/core-js-2.5.7.tgz#f972608ff0cead68b841a16a932d0b183791814e"
  integrity sha512-RszJCAxg/PP6uzXVXL6BsxSXx/B05oJAQ2vkJRjyjrEcNVycaqOmNb5OTxZPE3xa5gwZduqza6L9JOCenh/Ecw==

core-js@^2.5.7:
  version "2.6.1"
  resolved "https://nexus.wired-aws.com/repository/npm-group/core-js/-/core-js-2.6.1.tgz#87416ae817de957a3f249b3b5ca475d4aaed6042"
  integrity sha512-L72mmmEayPJBejKIWe2pYtGis5r0tQ5NaJekdhyXgeMQTpJoBsH0NL4ElY2LfSoV15xeQWKQ+XTTOZdyero5Xg==

core-js@^2.6.5:
  version "2.6.9"
  resolved "https://nexus.wired-aws.com/repository/npm-group/core-js/-/core-js-2.6.9.tgz#6b4b214620c834152e179323727fc19741b084f2"
  integrity sha512-HOpZf6eXmnl7la+cUdMnLvUxKNqLUzJvgIziQ0DiF3JwSImNphIqdGqzj6hIKyX04MmV0poclQ7+wjWvxQyR2A==
```
The downloaded package would be the one with version `2.6.1` which should be the final result outputed on the screen. If I understand correctly this modification will result with `3.1.4`, the highest possible version defined in the yarn.lock, being outputed

Reply:
This problem is unfortunately a lot less trivial 😢

Reply:
@celcius112 I changed the implementation. Now i check that the yarn line contains the version present in the package.json. On your example only `core-js@^2.5.7:` should match.